tests/helper/client_test.cpp

```diff
@@ -2212,4 +2212,55 @@ TEST(ClientTest, RequestExecDisabledClient)
     }
 }
 
+TEST(ClientTest, AsmDDRulesIsAddedWhenRulesFileIsEmpty)
```

Comment:
There are currently no engine or waf tests in which the ruleset is updated, only rule data, these are required as well.

Reply:
I'll do this as part of another PR and add tests for other listeners